### PR TITLE
Fixing client code

### PR DESCRIPTION
**Pull request by GunaKKIBM**
PR objective:
1.  Passing interface for fixing UTs.
2. Deprecating GC endpoints
3. Passing k8s client to fetch secret, config map

### CODE REVIEW

File: block/provider/provider.go [hunk 1]
```diff
@@ -70,61 +70,46 @@ type VPCBlockProvider struct {
...
+		conf.VPCConfig.EndpointURL = conf.VPCConfig.G2EndpointPrivateURL
+	} else {
+		conf.VPCConfig.EndpointURL = conf.VPCConfig.G2EndpointURL
+	}
 
-		conf.VPCConfig.APIKey = conf.VPCConfig.G2APIKey
-		conf.VPCConfig.ResourceGroupID = conf.VPCConfig.G2ResourceGroupID
+	// update iam based public toke exchange endpoint
+	conf.VPCConfig.TokenExchangeURL = conf.VPCConfig.G2TokenExchangeURL
+	conf.VPCConfig.APIKey = conf.VPCConfig.G2APIKey
```
Comment (GunaKKIBM) [inline block/provider/provider.go:88]: No need to init api key - add TODO

**Review by sameshai — APPROVED** [review]

File: block/provider/provider.go [hunk 2]
```diff
@@ -70,61 +71,47 @@ type VPCBlockProvider struct {
...
+	conf.VPCConfig.ResourceGroupID = conf.VPCConfig.G2ResourceGroupID
 
-		//Set API Generation As 2 (if unspecified in config/ENV-VAR)
-		if conf.VPCConfig.G2VPCAPIGeneration <= 0 {
-			conf.VPCConfig.G2VPCAPIGeneration = NEXTGenProvider
-		}
-		conf.VPCConfig.VPCAPIGeneration = conf.VPCConfig.G2VPCAPIGeneration
+	//Set API Generation As 2 (if unspecified in config/ENV-VAR)
+	if conf.VPCConfig.G2VPCAPIGeneration <= 0 {
```
Comment (arahamad) [inline block/provider/provider.go:94]: I think we need to make it G2 i.e next gen always as gc is not supported now

File: block/provider/provider.go [hunk 3]
```diff
@@ -70,61 +71,47 @@ type VPCBlockProvider struct {
...
+	if conf.VPCConfig.G2APIVersion != "" {
+		conf.VPCConfig.APIVersion = conf.VPCConfig.G2APIVersion
+	}
 
-		//set provider-type (this usually comes from the secret)
-		if conf.VPCConfig.VPCBlockProviderType != VPCNextGen {
-			conf.VPCConfig.VPCBlockProviderType = VPCNextGen
-		}
+	//set provider-type (this usually comes from the secret)
+	if conf.VPCConfig.VPCBlockProviderType != VPCNextGen {
```
Comment (arahamad) [inline block/provider/provider.go:105]: here as well, only support gen2 i.e g2

File: go.mod [hunk 4]
```diff
@@ -4,8 +4,7 @@ go 1.18
 
 require (
 	github.com/IBM-Cloud/ibm-cloud-cli-sdk v0.6.7
-	github.com/IBM/ibmcloud-volume-interface v1.1.1
-	github.com/IBM/secret-common-lib v1.1.1
+	github.com/IBM/ibmcloud-volume-interface v1.1.2-0.20230116094327-084d382598e9
```
Comment (arahamad) [inline go.mod:7]: update with proper release version

**Review by arahamad — COMMENTED** [review]
Can you address these comments

**Review by prankulmahajan — APPROVED** [review]

**Pull request merged**